packages/charts/react-charting/src/components/DeclarativeChart/PlotlySchemaAdapter.ts

```diff
@@ -218,16 +219,46 @@ const getSecondaryYAxisValues = (
   };
 };
 
+export const getSchemaColors = (colors: Array<string | number | null | undefined>): string[] | undefined => {
+  const hexColors: string[] = [];
+  if (!colors) {
+    return undefined;
+  }
+  if (isArrayOrTypedArray(colors)) {
+    colors.forEach((element: string | number | null | undefined) => {
+      if (element && element.toString().trim() !== '') {
+        const parsedColor = d3Color(element.toString());
+        if (parsedColor) {
```

Comment:
don't change the order of the colors. If some color comes out invalid, then inject fluent color to it.